packages/shared/src/child_bounties.ts

```diff
@@ -0,0 +1,2650 @@
+  )
+  await sendTransaction(proposeChildCuratorTx.signAsync(testAccounts.bob))
+
+  await client.dev.newBlock()
+
+  // Verify child bounty status is CuratorProposed
+  const proposedChildBounty = await getChildBounty(client, parentIndex, childIndex)
+  expect(proposedChildBounty.status.isCuratorProposed).toBe(true)
+
+  // Test 1: Curator rejects the assignment (unassigns themselves)
+  const unassignCuratorTx = client.api.tx.childBounties.unassignCurator(parentIndex, childIndex)
+  await sendTransaction(unassignCuratorTx.signAsync(testAccounts.charlie))
+
+  await client.dev.newBlock()
+
+  // Verify child bounty status is back to Added
+  const rejectedChildBounty = await getChildBounty(client, parentIndex, childIndex)
+  expect(rejectedChildBounty.status.isAdded).toBe(true)
+
+  // Test 2: Propose curator again and then cancel the child bounty
+  const proposeChildCuratorTx2 = client.api.tx.childBounties.proposeCurator(
+    parentIndex,
+    childIndex,
+    testAccounts.charlie.address,
+    childCuratorFee,
+  )
+  await sendTransaction(proposeChildCuratorTx2.signAsync(testAccounts.bob))
+
+  await client.dev.newBlock()
+
+  // Verify child bounty status is CuratorProposed again
+  const proposedChildBounty2 = await getChildBounty(client, parentIndex, childIndex)
+  expect(proposedChildBounty2.status.isCuratorProposed).toBe(true)
+
+  // Cancel child bounty by parent curator
+  const closeChildBountyTx = client.api.tx.childBounties.closeChildBounty(parentIndex, childIndex)
+  await sendTransaction(closeChildBountyTx.signAsync(testAccounts.bob))
+
+  await client.dev.newBlock()
+
+  // Check for ChildBountyCanceled event
+  await checkSystemEvents(client, { section: 'childBounties', method: 'Canceled' })
+    .redact({ redactKeys: /index|data/ })
+    .toMatchSnapshot('child bounty canceled events')
+
+  // Verify child bounty is removed from storage
+  const canceledChildBounty = await getChildBounty(client, parentIndex, childIndex)
+  expect(canceledChildBounty).toBeNull()
+
+  // Verify child bounty description is removed
+  const childBountyDesc = await getChildBountyDescription(client, parentIndex, childIndex)
+  expect(childBountyDesc).toBeNull()
+
+  // Verify child bounties count is 0
+  const childBountiesCount = await getParentChildBountiesCount(client, parentIndex)
+  expect(childBountiesCount).toBe(0)
+
+  await client.teardown()
+}
+
+/**
+ * Test: unassign curator different cases
+ *
+ * This test verifies the different curator unassignment scenarios to ensure that:
+ * - Self-unassignment by child curators refunds their deposit
+ * - Unassignment by parent curators slashes the child curator's deposit
+ * - Unauthorized unassignment attempts fail with appropriate errors
+ * - Different deposit handling based on who initiates the unassignment
+ *
+ * Test structure:
+ * 1. Create parent bounty and child bounty with active curator (Charlie)
+ * 2. Test child curator self-unassignment (should refund deposit)
```

Comment:
I believe these (2. through 4.) should be different tests - they're related, but can exist disjointed of one another.

The setup can be part of a helper, which is then called by each test.